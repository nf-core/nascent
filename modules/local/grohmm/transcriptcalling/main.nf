process GROHMM_TRANSCRIPTCALLING {
    tag "$meta.id"
    label 'process_high'
    label 'process_long'

    conda    (params.enable_conda ? "conda-forge::r-base=4.1.1 conda-forge::r-optparse=1.7.1 conda-forge::r-argparse=2.1.3 bioconda::bioconductor-genomicfeatures=1.46.1 bioconda::bioconductor-grohmm=1.28.0"  : null)
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-e9a6cb7894dd2753aff7d9446ea95c962cce8c46:0a46dae3241b1c4f02e46468f5d54eadcf64beca-0' :
        'quay.io/biocontainers/mulled-v2-e9a6cb7894dd2753aff7d9446ea95c962cce8c46:0a46dae3241b1c4f02e46468f5d54eadcf64beca-0' }"

    input:
    tuple val(meta), path(bams)
    path gtf
    path tuning_file

    output:
    tuple val(meta), path("*.transcripts.txt"), emit: transcripts
    tuple val(meta), path("*.eval.txt")       , emit: eval
    tuple val(meta), path("*.transcripts.bed"), emit: transcripts_bed
    tuple val(meta), path("*.tdFinal.txt")    , emit: td
    tuple val(meta), path("*.tdplot_mqc.jpg") , emit: td_plot
    path  "versions.yml"     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def tuning = tuning_file ? "--tuning_file ${tuning_file}" : ""
    """
    transcriptcalling_grohmm.R \\
        --bam_file ${bams} \\
        $tuning \\
        --outprefix ${prefix} \\
        --gtf $gtf \\
        --outdir ./ \\
        --cores $task.cpus \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        bioconductor-grohmm: \$(Rscript -e "library(groHMM); cat(as.character(packageVersion('groHMM')))")
    END_VERSIONS
    """
}
