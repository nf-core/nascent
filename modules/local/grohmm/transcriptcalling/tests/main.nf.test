nextflow_process {

    name "Test Process GROHMM_TRANSCRIPTCALLING"
    script "modules/local/grohmm/transcriptcalling/main.nf"
    process "GROHMM_TRANSCRIPTCALLING"

    test("Should fail without a tuning file") {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'mR1' ],
                    [ file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S0mR1.bam", checkIfExists: true),
                    file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S40mR1.bam", checkIfExists: true) ],
                    [],
                    [],
                ]
                input[1] = file(
                            "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/nascent/reference/kgChr7.gtf",
                            checkIfExists: true
                            )
                """
            }
        }

        then {
            assert process.failed
        }
    }

    test("Should run with a tuning file") {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id: 'mR1' ],
                    [ file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S0mR1.bam", checkIfExists: true),
                    file("https://raw.githubusercontent.com/Kraus-Lab/groHMM/master/inst/extdata/S40mR1.bam", checkIfExists: true) ],
                    [],
                    file("${projectDir}/tests/config/tuningparams_small.csv", checkIfExists: true),
                ]
                input[1] = file(
                            "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/nascent/reference/kgChr7.gtf",
                            checkIfExists: true
                            )
                """
            }
        }

        then {
            assert process.success
        }
    }

}
