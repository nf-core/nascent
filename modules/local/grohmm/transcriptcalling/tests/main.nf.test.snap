{
    "versions": {
        "content": [
            {
                "GROHMM_TRANSCRIPTCALLING": {
                    "r-base": "4.3.3",
                    "bioconductor-grohmm": "1.39.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-24T11:27:57.580852329"
    },
    "Should run with a tuning file": {
        "content": [
            [
                [
                    {
                        "id": "mR1"
                    },
                    "mR1.transcripts.txt:md5,06690e4932d6f2d597c2f482fb0848d2"
                ]
            ],
            [
                [
                    {
                        "id": "mR1"
                    },
                    "mR1.eval.txt:md5,4dc82be9170bf40cdb88e187d805f9df"
                ]
            ],
            [
                [
                    {
                        "id": "mR1"
                    },
                    "mR1.final.transcripts.bed:md5,db367d8482ee6b88af176c35e58807d9"
                ]
            ],
            [
                [
                    {
                        "id": "mR1"
                    },
                    "mR1.tdFinal.txt:md5,24d8292ec9f9ab8e2aac99fb6061d26c"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-24T11:27:57.184918004"
    }
}